### DISCUSSION TITLE
Chasing Gremlins - Suddenly my motor controllers won't  work

### DISCUSSION THREAD

**Discussion by Klippert** [grblHAL/core#553] [Q&A] [answered]
Hello,

This is probably more a request for help debugging, than an actual GRBLHAL problem, but I've gotten such excellent advice here before, that's it seems my best bet for some fresh inspiration.

I'm running GRBLHal on an Teensy with Phil's breakout board, and have been in the process of upgrading my motors from Closed-loop HBS57 to (expensive) Clearpath servos. (though for a start only on my gantried Y axis and the X axis.) - So I just wanted to test my installation on the Y axis, with the two servos, and I removed the two old drives, and connected the wiring to the clearpath servos, and it works. I've got the two new servos turning alright with step and dir control.

But for some reason, my old X axis motor and Z axis motor are now dead. And I'm just completely confused as to why. 
I checked power first, and the two drivers are powered both. They also light up Green. But I just can't control the motors. I firmly believe that the wiring for these two drives are the same as before. (though  evidently something is no longer working) 

![image](https://github.com/user-attachments/assets/91750936-4b33-4cec-ad2c-3a88d74ac6cb)

As one can see here above, the wiring there is based on a pull down design, however the breakout board:
![image](https://github.com/user-attachments/assets/79233d9f-40b4-4d26-9fcf-e8c91b58e9b1)
is using a pull up (or common ground) - however i've used that the las 3-4 years without any problems. Just having the PUL-, Dir- and ENA- connected to grnd on the board, and then the repective ´+´ to the outputs. 

When power up, I get green lights on the drives (meaning they are not in error, and got power.) - I've also measured the outputs of the breakout board, which for ie. Ena + Grnd measures 4,6v (which I guess is close enough to 5v to make a optocoupler trigger) 
I also measured the Dir + grnd, and I also get high and low signals (+4.5 and 0v) when i jog backwards and forwards.

As i had extra of the HBS57 drivers, I also tried replacing that, with no effect.

So my great question really is, what would you suggest testing next?
 
To me it seems like the drivers are not enabling correctly - but i can't understand why. And i have this funny feeling i'm going to be kicking myself when i find the error... ;-)

Best regards
and thanks in advance for any bright ideas
Kristoffer

**Comment by terjeio** [comment]
> To me it seems like the drivers are not enabling correctly

$4 setting is incorrect? Have you tried inverting the outputs for X and Z?

**Reply by Klippert** [reply]
Hello Terje, 

Thanks for the response. 
I've set it up for enabling all motors always and set enable to be "high". 
And when i stick a volt meter on Ground and EN i get 4.6v, so that should be high enough? 

But despite this output, it seems exactly as you say, they just don't enable. - maybe i should try just feeding 5v straight to the enable of the servo drives, and see if they will enable that way.... ?

/K

**Reply by terjeio** [reply, marked as answer]
> As one can see here above, the wiring there is based on a pull down design

Which means you have to set the enable output 0V to turn on the LED. Did you try inverting X and Z enable? Or is the input _stepper disable_?

**Reply by Klippert** [reply]
Hello Terje,

Ok, finally got some time to work on it again. It would seem that you were right. I accidentially powered the drives up without the controller connected to them, and they enabled. So i realised that contrary to documentation, they need enable low, to enable. and guess what, after setting it low in GRBLHal configuration, the two old drives work. now I just need to figure out why my enable outputsare all following the state of X. I think i remember something about the breakout board doing that with some jumpers. 
But i need my clearpath servos to enable with a high, and the old drivers to enable with a low.

Thanks again for your help, and as usual you were right, even if it was for different reasons :-) 

Best regards
Kristoffer

**Reply by Klippert** [reply]
Found it! - There is a board jumper that I have to cut to run seperate enables for each stepper. 
![image](https://github.com/user-attachments/assets/9b8e8d64-d274-4d90-a7ed-a2b37f1ad4b7)

**Reply by Klippert** [reply]
And it works. - Just to confirm it did the trick.
